larky/src/main/resources/vendor/ecdsa/keys.star

```diff
@@ -0,0 +1,336 @@
+        self.baselen = None
+        self.verifying_key = None
+        self.privkey = None
+        return self
+    self = __init__(_error__please_use_generate)
+
+    def _weierstrass_keygen(curve, entropy, hashfunc):
+        """Generate a private key on a Weierstrass curve."""
+        secexp = randrange(curve.order, entropy)
+        return self.from_secret_exponent(secexp, curve, hashfunc)
+    self._weierstrass_keygen = _weierstrass_keygen
+
+     def generate(curve=NIST192p, entropy=None, hashfunc=sha1):
+        """
+        Generate a random private key.
+        :param curve: The curve on which the point needs to reside, defaults
+            to NIST192p
+        :type curve: ecdsa.curves.Curve
+        :param entropy: Source of randomness for generating the private keys,
+            should provide cryptographically secure random numbers if the keys
+            need to be secure. Uses os.urandom() by default.
+        :type entropy: callable
+        :param hashfunc: The default hash function that will be used for
+            signing, needs to implement the same interface
+            as hashlib.sha1
+        :type hashfunc: callable
+        :return: Initialised SigningKey object
+        :rtype: SigningKey
+        """
+        # if isinstance(curve.curve, CurveEdTw):
+        #     return cls._twisted_edwards_keygen(curve, entropy)
+        return self._weierstrass_keygen(curve, entropy, hashfunc)
+    self.generate = generate
+    
+    def from_secret_exponent(secexp, curve=NIST192p, hashfunc=sha1):
+        """
+        Create a private key from a random integer.
+        Note: it's a low level method, it's recommended to use the
+        :func:`~SigningKey.generate` method to create private keys.
+        :param int secexp: secret multiplier (the actual private key in ECDSA).
+            Needs to be an integer between 1 and the curve order.
+        :param curve: The curve on which the point needs to reside
+        :type curve: ecdsa.curves.Curve
+        :param hashfunc: The default hash function that will be used for
+            signing, needs to implement the same interface
+            as hashlib.sha1
+        :type hashfunc: callable
+        :raises MalformedPointError: when the provided secexp is too large
+            or too small for the curve selected
+        :raises RuntimeError: if the generation of public key from private
+            key failed
+        :return: Initialised SigningKey object
+        :rtype: SigningKey
+        """
+        # currently we implement ecdsa only for secp256k1 which uses weierstrass curve. 
+        # if isinstance(curve.curve, CurveEdTw):
+        #     fail('ValueError("Edwards keys don\'t support setting the secret scalar (exponent) directly")')
+        self = __init__(_error__please_use_generate=True)
+        self.curve = curve
+        self.default_hashfunc = hashfunc
+        self.baselen = curve.baselen
+        n = curve.order
+        if not 1 <= secexp < n:
```

Comment:
```suggestion
        if not (1 <= secexp and  secexp < n):
```

Reply:
This pr is still in progress. I will fix many stuff like this. Thank you!